### PR TITLE
Facebook messenger

### PR DESCRIPTION
**Pull request by mebibou**
see #77

### CODE REVIEW

File: dist/angular-socialshare.js [hunk 1]
```diff
@@ -42,239 +42,239 @@
...
+          'source': '',
+          'caption': '',
+          'redirectUri': '',
+          'trigger': 'click',
+          'popupHeight': 300,
+          'popupWidth': 400
+        }
+      },
+      {
+        'provider': 'facebookMessenger',
```
Comment (45kb) [inline dist/angular-socialshare.js]: facebookMessenger `-> to ->` facebook-messenger 👍

File: dist/angular-socialshare.js [hunk 2]
```diff
@@ -514,6 +514,12 @@
           + ',top=' + ($window.innerHeight - attrs.socialsharePopupHeight) / 2 + ',left=' + ($window.innerWidth - attrs.socialsharePopupWidth) / 2);
       }
     }
+    , facebookMessengerShare = function facebookMessengerShare($window, $location, attrs, element) {
+
+      var href = 'fb-messenger://share?link=' + encodeURIComponent(attrs.socialshareUrl || $location.absUrl());
+
+      element.attr('href', href);
```
Comment (45kb) [inline dist/angular-socialshare.js]: This could not work on a `<button>`

**Comment by 45kb** [comment]
Great work, just look at my notes please :)

File: dist/angular-socialshare.js (hunk 2 above)
Comment (mebibou) [inline dist/angular-socialshare.js]: Yes, just like `whatsapp` and `viber`, they can only work on a link that will trigger the app to open.

Comment (45kb) [inline dist/angular-socialshare.js]: Ok, so, the only way is to create an hidden `<a href="viber://"></a>` element in JS and trigger click on it i guess

Comment (45kb) [inline dist/angular-socialshare.js]: Not even this works, i tried with an hidden <a> and triggering click() on it doesn't work, seems really like it only works if it's an href="" element only and visible in page :(

**Comment by 45kb** [comment]
Going to merge and make some change, thanks!

**Comment by 45kb** [comment]
shipped the new versoin 📦  https://github.com/720kb/angular-socialshare/releases/tag/2.1.2

Thanks a lot Guillaume!

**Pull request merged**